docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,707 @@
+-0.61,-0.37,-0.12,0.55,-1.00,0.84,-0.02,1.30,-0.24,-0.50,-2.12,148.12
+-0.85,-0.91,1.81,0.02,-0.78,-1.41,-1.09,-0.65,0.90,-0.37,-0.22,402.20
+0.28,1.05,-0.24,0.30,-0.99,0.19,0.32,-0.95,-1.19,-0.63,0.75,443.51
+```
+
+Reading this file using `TextLoader`:
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Create the reader: define the data columns and where to find them in the text file.
+var reader = TextLoader.CreateReader(env, ctx => (
+        // We read the first 11 values as a single float vector.
+        FeatureVector: ctx.LoadFloat(0, 10),
+        // Separately, read the target variable.
+        Target: ctx.LoadFloat(11)
+    ),
+    // Default separator is tab, but we need a comma.
+    separator: ',');
+
+
+// Now read the file (remember though, readers are lazy, so the actual reading will happen when the data is accessed).
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+
+If the schema of the data is not known at compile time, or too cumbersome, you can revert to the dynamically-typed API: 
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Create the reader: define the data columns and where to find them in the text file.
+var reader = new TextLoader(env, new TextLoader.Arguments
+{
+    Column = new[] {
+	    // We read the first 10 values as a single float vector.
+        new TextLoader.Column("FeatureVector", DataKind.R4, new[] {new TextLoader.Range(0, 9)}),
+        // Separately, read the target variable.
+        new TextLoader.Column("Target", DataKind.R4, 10)
+    },
+    // Default separator is tab, but we need a comma.
+    Separator = ","
+});
+
+// Now read the file (remember though, readers are lazy, so the actual reading will happen when the data is accessed).
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+## How do I look at the intermediate data?
+
+Oftentimes, when we construct the experiment, we want to make sure that the data processing 'up to a certain moment' produces the results that we want. With ML.NET it is not very easy to do: since all ML.NET operations are lazy, the objects we construct are just 'promises' of data.
+
+We will need to create the cursor and scan the data to obtain the actual values. One way to do this is to use [schema comprehension](SchemaComprehension.md) and map the data to an `IEnumerable` of user-defined objects.
+
+Another mechanism that lets you inspect the intermediate data is the `GetColumn<T>` extension method. It lets you look at the contents of one column of your data in a form of an `IEnumerable`.
+
+Here is all of this in action:
+
+Example file (https://github.com/dotnet/machinelearning/blob/master/test/data/adult.tiny.with-schema.txt):
```

Comment:
>https://github.com/dotnet/machinelearning/blob/master/test/data/adult.tiny.with-schema.txt [](start = 14, length = 90)

These are files within the same repro where we're putting the files. It would probably be better practice if these were [relative links](https://blog.github.com/2013-01-31-relative-links-in-markup-files/).